### ISSUE TITLE
influxdb_return.py throws exception

### ISSUE THREAD

**Issue by bewing** [saltstack/salt#40937]
### Description of Issue/Question
Attempting to use --return influxdb with test.ping results in an exception in the minion log, and no data written to influxdb
### Setup
```bash
# cat /etc/salt/master | grep influxdb
influxdb.host: 'influxdb'
influxdb.db: 'metrics'
influxdb.user: 'writer'
influxdb.password: 'writer'
influxdb.port: 8086
```

### Steps to Reproduce Issue
```bash
# salt --return influxdb master test.ping
master:
    True
```
```python
2017-04-27 19:35:06,141 [salt.minion      ][INFO    ][23662] User root Executing command test.ping with jid 20170427193506130542
2017-04-27 19:35:06,141 [salt.minion      ][DEBUG   ][23662] Command details {'tgt_type': 'glob', 'jid': '20170427193506130542', 'tgt': 'master', 'ret': 'influxdb', 'user': 'root', 'arg': [], 'fun': 'test.ping'}
2017-04-27 19:35:06,157 [salt.minion      ][INFO    ][28564] Starting a new job with PID 28564
2017-04-27 19:35:06,181 [salt.utils.lazy  ][DEBUG   ][28564] LazyLoaded test.ping
2017-04-27 19:35:06,182 [salt.utils.lazy  ][DEBUG   ][28564] LazyLoaded direct_call.get
2017-04-27 19:35:06,182 [salt.minion      ][DEBUG   ][28564] Minion return retry timer set to 9 seconds (randomized)
2017-04-27 19:35:06,182 [salt.minion      ][INFO    ][28564] Returning information for job: 20170427193506130542
2017-04-27 19:35:06,183 [salt.transport.zeromq][DEBUG   ][28564] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'master', 'tcp://127.0.0.1:4506', 'aes')
2017-04-27 19:35:06,183 [salt.crypt       ][DEBUG   ][28564] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'master', 'tcp://127.0.0.1:4506')
2017-04-27 19:35:06,209 [salt.utils.lazy  ][DEBUG   ][28564] LazyLoaded influxdb.returner
2017-04-27 19:35:06,210 [salt.minion      ][ERROR   ][28564] The return failed for job 20170427193506130542 sequence item 1: expected string, int found
2017-04-27 19:35:06,210 [salt.minion      ][ERROR   ][28564] Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1556, in _thread_return
    minion_instance.returners[returner_str](ret)
  File "/usr/lib/python2.7/dist-packages/salt/returners/influxdb_return.py", line 152, in returner
    serv = _get_serv(ret)
  File "/usr/lib/python2.7/dist-packages/salt/returners/influxdb_return.py", line 130, in _get_serv
    version = _get_version(host, port, user, password)
  File "/usr/lib/python2.7/dist-packages/salt/utils/decorators/__init__.py", line 248, in _memoize
    args_ = ','.join(list(args) + ['{0}={1}'.format(k, kwargs[k]) for k in sorted(kwargs)])
TypeError: sequence item 1: expected string, int found
```

### Versions Report
```
Salt Version:
           Salt: 2017.5.0-2-gfa2c8f8

Dependency Versions:
           cffi: 1.10.0
       cherrypy: Not Installed
       dateutil: 2.6.0
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.6 (default, Oct 26 2016, 20:30:19)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-112-generic
         system: Linux
        version: Ubuntu 14.04 trusty

```

**Comment by Ch3LL** [comment]
Looks like i'm able to replicate this. This is the first bad commit: 79f2dc6fa1bea78888fd2b3bf4b52094cd6e4f9b which is this pr https://github.com/saltstack/salt/pull/37067

This error shows up in develop and nitrogen branches. 

When bringing up pudb to debug I show this in args;

```
>>> list(args)
['172.17.0.1', 8086, 'test', 'password']
```

```
In [1]: args = ('172.17.0.1', 8086, 'test', 'password')

In [2]: ','.join(list(args))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-0509b6e523ad> in <module>()
----> 1 ','.join(list(args))

TypeError: sequence item 1: expected string, int found
```

So the port number is the int type 8086. We will need to fix this thanks.

**Comment by bewing** [comment]
Putting quotes around the port does appear to resolve this:
`salt --return influxdb --return_kwargs '{"host": "localhost", "db": "metrics", "user": "writer", "password": "writer", "port": "8086"}' master test.ping`
does not throw the memoize error.

**Comment by Ch3LL** [comment]
I think @cachedout has a solution to this issue. Mind submitting a pr @cachedout ?

**Comment by cachedout** [comment]
Sure. I just have to remember where I put that diff I sent you. ;]

**Comment by cachedout** [comment]
OK, #41014 should do it.

**Comment by cachedout** [comment]
Closed via #41014
